### ISSUE TITLE
issues about <07_convolution_nn.ipynb>

### ISSUE THREAD

**Issue by Johnson-yue** [madalinabuzau/tensorflow-eager-tutorials#1]
Hello, thank you for sharing your code. It help me a lot for learning tensorflow eager model. But  I have some question in __07.ipynb__ . 
"""
In[13]:  class EmotionRecognitionCNN(tf.keras.Model):
          .........
          .........
                def fit(......):
 line 178:         for images, target in tfe.Iterator(training_data):
     |
line 183:          for images, target in tfe.Iterator(training_data):
"""

Why not merge the two for loop？？

**Comment by madalinabuzau** [comment]
Hi there :)! Thank you so much for your feedback. I really appreciate it.

With regards to your question: we have to iterate again through the training data to compute the loss after performing gradient descent for 1 epoch. If we merge the two for loops the loss will be computed whilst the network is still training. The losses for the first batches will be higher as the network still hasn't seen the entire data. To compute the loss after training for one epoch, we have to wait until the network has seen all batches and then compute the final loss. You can merge the for loops in any case, if you're looking for speed improvements. 

Hope this helps and if you need any help please let me know!

**Comment by Johnson-yue** [comment]
Oh， I know，you think the losses for the first batches was unreasonable。

[Question 2]: Do you use __Tensorboard__ ? 
Would you add tutorial for tensorboard? or  data aug for image using tensorflow. (Maybe _ tf.data_ ?) thank you!!

**Comment by madalinabuzau** [comment]
[Question 1]: this represents a problem only if you want to compute the loss after each epoch. You can easily combine the two for loops and then have the loss plotted as a function of 'Number of batches' seen instead of 'Number of epochs'. 

[Question 2]: Yes, I do use Tensorboard but I don't use it with Eager mode.  I don't think that at the moment you can visualize the graph when you're using Eager mode. I'll check soon to see the features available for this mode and make a tutorial on it :). 

Thanks for your interest and good luck improving your ML skills :)!

**Comment by Johnson-yue** [comment]
Thank you for your reply and look forward to your  tutorial update。
